carbon-route.html

```diff
@@ -271,48 +276,73 @@
         tailPath = '/' + tailPath;
       }
 
-      this._skipMatch = true;
       this._matched = matched;
-      this.data = namedMatches;
+
       var tailPrefix = this.route.prefix + matched;
 
-      if (!this.tail || this.tail.prefix !== tailPrefix || this.tail.path !== tailPath) {
-        this.tail = {
+
+      // HACK(rictic): skirting around 1.0's lack of a setMulti by poking at
+      //     internal data structures. I would not advise that you copy this
+      //     example.
+      //
+      //     This is juggling footguns.
+      this._propertySetter('data', namedMatches);
+
+      var tailChanged =
+          !this.tail ||
+          this.tail.prefix !== tailPrefix ||
+          this.tail.path !== tailPath;
+      if (tailChanged) {
+        this._propertySetter('tail', {
           prefix: tailPrefix,
           path: tailPath,
           __queryParams: this.route.__queryParams
-        };
+        });
+      }
+
+      this._dataInUrl = {};
+      for (var key in namedMatches) {
+        this._dataInUrl[key] = namedMatches[key];
       }
+
       this._setActive(true);
-      this._skipMatch = false;
+      if (tailChanged) {
+        this._pathEffector('tail', this.tail);
+        this._notifyPathUp('tail', this.tail);
+      }
+      this._pathEffector('data', this.data);
+      this._notifyPathUp('data', this.data);
```

Comment:
I'll leave it to your judgement, although it has to be said that this method is currently out of sorts (in terms of conceptual clutter), and will most likely be impenetrable to an uninformed contributor looking upon it for the first time.